src/modules/resource/status.c

```diff
@@ -442,6 +463,23 @@ static void sched_status_cb (flux_t *h,
         flux_log_error (h, "error responding to resource.sched-status");
 }
 
+/* Watch for resource eventlog events that might invalidate cached data.
+ * That is pretty much all of them:
+ * resource-define - could be called in test from 'flux resource reload'
+ * resource-update - expiration only at this time but think of the future
+ * online, offline - node went up/down
+ * drain, undrain - node was drained/udrained
+ */
+static void reslog_cb (struct reslog *reslog,
+                       const char *name,
+                       json_t *context,
+                       void *arg)
+{
+    struct status *status = arg;
+
+    invalidate_cache (&status->cache);
+}
```

Comment:
In perf tracing, creating the `all` resource set was the most expensive operation, and `R_all` should only be invalidated by the first two events in the list above. Would it add too much complexity to save `R_all` unless a `resource-define` or `resource-update` event is seen? (you could perhaps add an optional event name parameter to `invalidate_cache`?)

On a very large system I fear the `online` `offline` `drain` and `undrain` events will be so frequent that we won't see as much benefit from the caching as we might otherwise.

Reply:
Sure, I'll do that.